src/components/csv-download-button.tsx

```diff
@@ -0,0 +1,59 @@
+'use client';
+
+import { useFilterState } from '@/app/common/hooks/use-filter-state';
+import { CSVDownloadObjectType, getTxnData } from '@/common/utils/csv-download-utils';
+import { Box, Icon } from '@/ui/components';
+import { CSVLink } from 'react-csv';
+import { Caption } from '@/ui/typography';
+import React, { memo, useRef, useState } from 'react';
+import { FiDownload } from 'react-icons/fi';
+import { useQueryClient } from 'react-query';
+
+import { useRouter } from 'next/router';
+
+export const CSVDownloadButton = memo(() => {
+  const [transactionData, setTransactionData] = useState<CSVDownloadObjectType[]>([]);
+  const queryClient = useQueryClient();
+  const { activeFilters } = useFilterState();
+  const {
+    query: { principal },
+  } = useRouter();
+  const csvLink = useRef<CSVLink & HTMLAnchorElement & { link: HTMLAnchorElement }>(null);
+
+  const donwloadCSV = () => {
+    const formattedTxnData = getTxnData(queryClient, activeFilters, principal as string);
+
+    setTransactionData(formattedTxnData);
+    console.log(formattedTxnData[0]);
+
+    // This is added to prevent immediate invocation of click function.
+    // Click should get invoked after the setState render is finished
+    setTimeout(() => {
+      if (csvLink.current && csvLink.current.link) csvLink.current.link.click();
+    }, 0);
+  };
+
+  return (
+    <Box position="relative" marginLeft={'auto'} alignSelf={'center'}>
+      <Caption
+        display="flex"
+        alignItems="center"
+        _hover={{ cursor: 'pointer', color: 'textTitle' }}
+        data-test="csv-download-button"
+        position="relative"
+        zIndex={999}
```

Comment:
Could you please consider removing zIndex? or use the lowest value needed.

Reply:
This code I referred from the FilterButton component file as CSVDownload button would be added adjacent to it.
![image](https://github.com/hirosystems/explorer/assets/20265606/efc61ef7-4745-46a7-938a-b1c68058d3d6)

If you want, I can check which lowest value of ZIndex would work. Please let me know how to proceed

Reply:
If you rebase to main you'll get the updated zIndexs. We recently refactored them across the UI.

Reply:
Sure, I'll check that